src/java/boa/compiler/visitors/CodeGeneratingVisitor.java

```diff
@@ -51,6 +52,9 @@
 	 */
 	String identifier="";
 	String traversalNodeIdentifier="";
+	String funcAssignParameter="";
```

Comment:
Is it possible to pull the renaming logic out into a transformer, and then just call that transformer here?  That would make it a lot cleaner and modular.